pytorch_lightning/plugins/training_type/ddp.py

```diff
@@ -450,7 +448,6 @@ def reconciliate_processes(self, trace: str):
 
         for pid in self._pids:
             if pid != os.getpid():
-                os.kill(pid, signal.SIGTERM)
-                # os.kill(pid, signal.SIGKILL)
-            shutil.rmtree(sync_dir)
-            raise DeadlockDetectedException(f"DeadLock detected from rank: {self.global_rank} \n {trace}")
+                os.kill(pid, signal.SIGKILL)
+        shutil.rmtree(sync_dir)
+        raise DeadlockDetectedException(f"DeadLock detected from rank: {self.global_rank} \n {trace}")
```

Comment:
do we have the prior exception context to `raise Exception from e` ?

Reply:
Yes, it is passed as the trace.